nixos/modules/security/acme/default.nix

```
@@ -1113,6 +1136,17 @@ in
                 used for variables suffixed by "_FILE".
               '';
             }
+
+            {
+              assertion = lib.all (
+                certOpts:
+                (certOpts.csr == null && certOpts.csrKey == null)
+                || (certOpts.csr != null && certOpts.csrKey != null)
+              ) certs;
+              message = ''
+                Either both `security.acme.certs.${cert}.csr` and `security.acme.certs.${cert}.csrKey` need to be specified, or null.
```

```suggestion
                When passing a certificate signing request both `security.acme.certs.${cert}.csr` and `security.acme.certs.${cert}.csrKey` need to be set.
```

I think this is a bit clearer, since when one of them is set the intent is probably to configure a CSR.